index.html

```diff
@@ -129,56 +129,478 @@ <h2 data-dft-for="">
+            };
+  
+            dictionary PromptResponseObject {
+              AppBannerPromptOutcome userChoice;
+            };
+  
+            enum AppBannerPromptOutcome {
+              "accepted",
+              "dismissed"
+            };
+          </pre>
         <p>
-          The [=note_taking=] <code>new_note_url</code> member is a <a>URL</a>
-          [=manifest/within scope=] of the manifest, to open when the user
-          wants to take a new note using the web application (e.g., from an
-          operating system shortcut icon or keyboard shortcut). Defining this
-          member indicates that the web application is a note-taking
-          application.
+          The <a>BeforeInstallPromptEvent</a> is dispatched when the site is
+          allowed to present a <a>site-triggered install prompt</a>, or prior
+          to the user agent presenting an <a>automated install prompt</a>. It
+          allows the site to cancel the <a>automated install prompt</a>, as
+          well as manually present the <a>site-triggered install prompt</a>.
+        </p>
+        <div class="note">
+          If the <a>BeforeInstallPromptEvent</a> is <em>not</em> cancelled, the
+          user agent is allowed to <a>present an install prompt</a>
+          (specifically, an <a>automated install prompt</a>) to the end-user.
+          Canceling the default action (via {{Event/preventDefault()}})
+          prevents the user agent from <a>presenting an install prompt</a>. The
+          user agent is free to run <a>steps to notify that an install prompt
+          is available</a> again at a later time.
+        </div>
+        <p data-dfn-for="PromptResponseObject">
+          The <dfn>PromptResponseObject</dfn> contains the result of calling
+          {{BeforeInstallPromptEvent\prompt()}}. It contains one member,
+          <dfn data-link-for="PromptResponseObject">userChoice</dfn>, which
+          states the user's chosen outcome.
         </p>
+        <p>
+          An instance of a <a>BeforeInstallPromptEvent</a> has the following
+          internal slots:
+        </p>
+        <dl>
+          <dt>
+            <dfn>[[\didPrompt]]</dfn>
+          </dt>
+          <dd>
+            A boolean, initially <code>false</code>. Represents whether this
+            event was used to <a>present an install prompt</a> to the end-user.
+          </dd>
+          <dt>
+            <dfn>[[\userResponsePromise]]</dfn>
+          </dt>
+          <dd>
+            A promise that represents the outcome of <a>presenting an install
+            prompt</a>.
+          </dd>
+        </dl>
+        <section>
+          <h4>
+            <code>prompt()</code> method
+          </h4>
+          <p>
+            The <dfn>prompt</dfn> method, when called, runs the following
+            steps:
+          </p>
+          <ol>
+            <li>If <var>this</var>.<a>[[\userResponsePromise]]</a> is pending:
```

Comment:
Same for any others: 
```suggestion
            <li>If <var>this</var>.{{BeforeInstallPromptEvent/[[userResponsePromise]]}} is pending:
```